### DISCUSSION TITLE
Streamlining dashboard filter queries

### DISCUSSION THREAD

**Discussion by rdauncey53** [grafana/grafana#45819] [Dashboards, Panels & Visualizations]
We use Grafana for some dashboards with quite a lot of data, and some of our dashboards haves 5-7 filters available on them. We have found that the loading time for the dashboard, i.e. when it is in this state:
![image](https://user-images.githubusercontent.com/98583473/155491670-3c2bf134-19ad-4966-814b-168d81b8abea.png)
can be very long - over a minute in some cases.

This appears to be the result of the queries run to get the filter labels; for each filter, the query `label_values(up, <label>)` is run, essentially querying every running container. Doing 5+ of these in a row really slows the dashboard down.

Would it be possible to add functionality to get all of the filter labels we want from one query? For example, at the moment if we want `label1` and `label2` as filters, we would have to include:
```
...
        "datasource": "Prometheus",
        "definition": "label_values(up, label1)",
        "description": null,
        "error": null,
        "hide": 0,
        "includeAll": true,
        "label": "Label 1",
        "multi": true,
        "name": "labelone",
        "options": [],
        "query": {
          "query": "label_values(up, label1)",
          "refId": "StandardVariableQuery"

... < other configuration > ...

        "datasource": "Prometheus",
        "definition": "label_values(up, label2)",
        "description": null,
        "error": null,
        "hide": 0,
        "includeAll": true,
        "label": "Label 2",
        "multi": true,
        "name": "labeltwo",
        "options": [],
        "query": {
          "query": "label_values(up, label2)",
          "refId": "StandardVariableQuery"
...
```
So we would have two large queries running, one for each label. Ideally, we would have functionality that allows us to run one query e.g.
`label_values(up, label1, label2, ...)`

I think this was partially addressed in #29984 but this is inactive and marked as low priority.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
